arbor/fvm_layout.cpp

```diff
@@ -592,31 +592,32 @@ fvm_mechanism_data fvm_build_mechanism_data(const cable_cell_global_properties&
         assign(param_names, util::keys(info.parameters));
         sort(param_names);
 
-        for (auto& p: param_names) {
+        std::size_t n_param = param_names.size();
+
+        for (std::size_t i = 0; i<n_param; ++i) {
+            const auto& p = param_names[i];
```

Comment:
The original `for (const auto& p: param_names)` (with added `const`) is a bit clearer methinks.

The `n_param` could be used to `reserve` capacity in `config.param_values` and `param_dflt`.

Reply:
I'll keep the `n_param` (because of parallel iteration through multiple containers of the same size). The reserve is definitely warranted; will add now.